### DISCUSSION TITLE
Running on Intel GPUs

### DISCUSSION THREAD

**Discussion by kris-rowe** [CliMA/Oceananigans.jl#2628] [Experimental features]
Is there interest in having support for running on Intel GPUs? 
This could be done using the Julia [OpenCL](https://github.com/JuliaGPU/OpenCL.jl) or [oneAPI](https://github.com/JuliaGPU/oneAPI.jl) packages.

I am willing to help with development and benchmarking on this front.

**Comment by glwagner** [comment]
Does KernelAbstractions support Intel GPUs? If its supported by KernelAbstractions then support can be developed for Oceananigans. There's some work in progress for AMD GPUs now on #2540.

One challenge is finding resources for continuous integration testing.

**Reply by kris-rowe** [reply]
It seems like KernelAbstractions is in the process of adding oneAPI support. (See [this PR](https://github.com/JuliaGPU/KernelAbstractions.jl/pull/312))

On the CI front, it should be easy to verify builds work since the oneAPI toolkits can be checkout using apt or similar.
For a complete CI pipeline, I could chat with some contacts at Intel to see if it is possible to launch jobs to their [DevCloud](https://www.intel.com/content/www/us/en/developer/tools/devcloud/overview.html).

**Reply by francispoulin** [reply]
Sounds like fun!

**Reply by glwagner** [reply]
> On the CI front, it should be easy to verify builds work since the oneAPI toolkits can be checkout using apt or similar.

But we need the hardware, right?

> I could chat with some contacts at Intel to see if it is possible to launch jobs to their [DevCloud](https://www.intel.com/content/www/us/en/developer/tools/devcloud/overview.html).

That could be interesting!

**Comment by kris-rowe** [comment]
Linking this PR comment since it is a relevant data point:
https://github.com/CliMA/Oceananigans.jl/pull/3475#issuecomment-2784665127

**Comment by kris-rowe** [comment]
There is a group at Argonne investigating the performance of Oceananigans on the Aurora supercomputer in case there are others interested in running on Intel GPUs as well.

**Reply by glwagner** [reply]
Curious to hear how that goes and if it makes sense to try to get an allocation on Aurora!

**Reply by francispoulin** [reply]
I am very interested to hear how this goes and happy to help set things up and even run things if you like.
